### ISSUE TITLE
[no-floating-promises] Suppress no-floating-promises rule with void operator

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#744]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-floating-promises": "error"
  }
}
```

```TS
void exampleReturningPromise();
```

**Expected Result**

No error.

**Actual Result**

Error: `Promises must be handled appropriately`

**Additional Info**

Motivation:

* It works in TSLint.
* It's practically *suggested* by TypeScript developers (see [here](https://github.com/microsoft/TypeScript/issues/13376#issuecomment-273289748) and [here](https://github.com/microsoft/TypeScript/issues/13376#issuecomment-274662923)).
* Other developers are using it for this purpose (see [here](https://github.com/Microsoft/TypeScript/issues/13376#issuecomment-478020804)).
* It's a lot more terse. Compare:
  ```ts
  // eslint-disable-next-line @typescript-eslint/no-floating-promises
  example();
  ```

  with:

  ```ts
  void example();
  ```
* It's easy [to grep for](https://github.com/typescript-eslint/typescript-eslint/pull/495#issuecomment-493515751), *if it's only being used for this purpose*.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
I ranted a bit, so TL;DR - I won't block this if someone wants to add it _as an option which is default off_.
I won't ever work on it myself, but I'll review a PR for it.

---

I still don't agree with it.

It's terse, yes, but terse !== better. Back in the days of old school programming where every byte of source code mattered - terse was better. Now a days with IDE tooling being as awesome as it is, you should aim for clarity in your code.

ESLint disable comments are super clear to anyone and everyone. It requires no extra thinking to notice, and the intention is instantly very clear in a github PR review. And is super easy to grep for.

The void operator is small, easy to miss, requires education to understand its meaning. It's only easy to grep for if it is the only way the void operator is used, as you point out.

IDE plugins for eslint (such as the vscode plugin) make it very easy to add the comment - you can do it in with two clicks when the error is detected.

Finally, just because the typescript devs suggest it, doesn't mean it's the best way to do things. You are misunderstanding the context with which they are suggesting it. You're reading it as a user of the language, when you should be reading it as a language designer.

They are suggesting it as a way to signal to the compiler that the promise is consumed, because the compiler doesn't have disable comments (well it has `@ts-ignore`, but that's a sledgehammer when you need a nutcracker).
They are suggesting it because they don't want to add a new operator for the sole purpose of consuming promises. Which makes sense from the POV of a language designer - why add a brand new keyword which only exists within typescript, when one that does the same thing exists?

**Comment by prokopsimek** [comment]
👍
